alchemist-graphql/src/commonMain/kotlin/it/unibo/alchemist/boundary/graphql/client/DefaultGraphQLClient.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2010-2023, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.graphql.client
+
+import com.apollographql.apollo3.ApolloCall
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.api.Mutation
+import com.apollographql.apollo3.api.Query
+import com.apollographql.apollo3.api.Subscription
+import com.apollographql.apollo3.network.ws.GraphQLWsProtocol
+import com.apollographql.apollo3.network.ws.WebSocketNetworkTransport
+
+/**
+ * Default GraphQL client implementation.
+ * @param client the Apollo client
+ */
+class DefaultGraphQLClient(private val client: ApolloClient) : GraphQLClient {
```

Comment:
This is a strange piece of engineering: the outer class knows its builder. Can you elaborate on the design choice?

Reply:
The reason behind this design was to enforce the use use of builder, much like Apollo Client API make their clients being built. If this is bad, I will change it in a standard builder.

Reply:
A classic (non step-) Builder serves two main purposes:
1. support for many parameters with a default
2. provide a clear indication of what some parameter does

It is thus particularly useful when the target language has no default and named parameter (like Java) and a bit less useful when the language does (Kotlin, Scala). Is Apollo a Java library? Or is it Kotlin-native?

Reply:
Apollo is written mainly Kotlin, but I got your point and so the Builder isn't really necessary; I can remove the builder and build it through the factory directly.

Reply:
I removed the builder and use only named parameters and the factory.